File: qe-selenium/qe-selenium-python/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as Ex
from selenium.webdriver.common.action_chains import ActionChains

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/keyboard-events/")
    wait=WebDriverWait(driver,10)
    actions=ActionChains(driver)
    print(driver.title)
    # driver.find_element(By.XPATH,"/html/body/div/main/div/div/div/div/div[2]/h1").send_keys("hi")
    driver.find_element(By.XPATH,"/html/body/div/main/div/div/div/div/div[2]/h1").click()
    actions.send_keys("hi").perform()
    print(driver.find_element(By.XPATH,"/html/body/div/main/div/div/div/div/div[2]/h1").text)
    
    driver.quit



# from selenium import webdriver
# from selenium.webdriver.common.by import By
# from selenium.webdriver.support.ui import WebDriverWait
# from selenium.webdriver.support import expected_conditions as EC
# from selenium.webdriver.common.action_chains import ActionChains

# driver = webdriver.Firefox()
# driver.get("https://training-support.net/webelements/keyboard-events/")

# wait = WebDriverWait(driver, 10)
# actions = ActionChains(driver)

# print(driver.title)

# # Click anywhere on the page to focus
# driver.find_element(By.TAG_NAME, "body").click()

# # Send keyboard input
# actions.send_keys("hi").perform()

# driver.quit()

File: qe-selenium/qe-selenium-java/src/main/java/com/demos/Activity9.java
package com.demos;

import org.openqa.selenium.By;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public abstract class Activity9 {
    public static void main(String[] args) {
        FirefoxDriver driver=new FirefoxDriver();
        Actions actions = new Actions(driver);

        driver.get("https://training-support.net/webelements/keyboard-events/");
        driver.getTitle();
        
        
       actions.click(driver.findElement(By.xpath("/html/body/div/main/div/div/div/div/div[2]/h1"))) 
            .sendKeys("hi")
            .perform();
        driver.quit();
        
    }
}
